From 3d8557c4c9ce19184cd02fa29a3ae8b4f7467c6f Mon Sep 17 00:00:00 2001
From: Yaroslav Zviezda <acroreiser@gmail.com>
Date: Sun, 3 Dec 2023 02:44:57 +0300
Subject: [PATCH 17/17] renderengine: gles: unconditionally skip
 PostRenderCleanup

Ultra-legacy targets works better without this.

Deprecates: https://github.com/LineageOS-UL/android_frameworks_native/commit/edd7c92d908590ea035b224d428b617661db535d
  https://github.com/LineageOS-UL/android_frameworks_native/commit/2456bab333f9fdb4e8d1ced7b88b835ec1ba756d
  https://github.com/LineageOS-UL/android_frameworks_native/commit/dff705049de26e5a6aa8ee1bbcc8e55c123981ec
Change-Id: I7974628d1207672fb3b6ad91f0bb4638b7584f7c
(cherry picked from commit 5255e5d2558c6160897c0b6e261aa0777ee82841)
---
 libs/renderengine/gl/GLESRenderEngine.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libs/renderengine/gl/GLESRenderEngine.cpp b/libs/renderengine/gl/GLESRenderEngine.cpp
index a512b9aca1..f47e27e4d6 100644
--- a/libs/renderengine/gl/GLESRenderEngine.cpp
+++ b/libs/renderengine/gl/GLESRenderEngine.cpp
@@ -970,8 +970,9 @@ void GLESRenderEngine::unbindFrameBuffer(Framebuffer* /*framebuffer*/) {
 }
 
 bool GLESRenderEngine::canSkipPostRenderCleanup() const {
-    return mPriorResourcesCleaned ||
-            (mLastDrawFence != nullptr && mLastDrawFence->getStatus() != Fence::Status::Signaled);
+//    return mPriorResourcesCleaned ||
+//            (mLastDrawFence != nullptr && mLastDrawFence->getStatus() != Fence::Status::Signaled);
+      return true;
 }
 
 void GLESRenderEngine::cleanupPostRender() {
-- 
2.48.1

